docs/development/communication.md

```diff
@@ -1,11 +1,147 @@
+
+The CAN Standard itself defines only the physical and data-link layer (layer 1 and 2 in OSI) with a maximum size of 8 bytes. In order to send larger messages, a protocol like [ISO 15765-2 - ISO-TP](https://en.wikipedia.org/wiki/ISO_15765-2) can be used.
+
+CAN offers a transmission speed up to 5 MBits/s, which greatly decreases the possible wire length. With a speed of 500 KBits/s, wires up to 100m can be used.
+
+#### Message Layout
+
+There are two frame formats defined, the base frame format with 11 identifier bits and the extended frame format, using 29 bits for message identification [2].
+
+The (simplified) base frame format:
+
+| #Bits |  1  | 11 |  7  | 0 .. 64 |  16 |  2  |  7  |
+|-------|:---:|:--:|:---:|:-------:|:---:|:---:|:---:|
+|       | SOF | ID | CTL |   DATA  | CRC | ACK | EOF |
+
+**SOF:** Start of frame <br />
+**ID:** Unique Identifier which also determines the priority <br />
+**CTL:** Control bits, including the length of data (0..8 bytes) <br />
+**CRC:** Cyclic Redundancy Check, used to detect transmission errors <br />
+**ACK:** Acknowledgment bits <br />
+**EOF:** Bit sequence to signal the end of frame
+
+With the extended frame format, there are additionaly 18 bits used for the ID field which can be used to split the ID into message priority and target and source address.
+
+#### Arbitration
+
+Since CAN is masterless, every node on the bus can start sending at any time. To detect collisions, the transceiver always compares its output to the input. If another device with a higher priority (lower ID) is sending at the same time, a difference between input and output will be detected and the lower priority device will retreat.
+
+### UART
+
+The Universal Asynchronous Receiver Transmitter is the most common serial communication used in MCUs. It has a receive (RX) and a transmit (TX) line, which are cross-connected between two devices. It allows only two connected devices and does not need any additional wiring, both microcontrollers should share the same ground and the same maximum voltage.
+
+Since there are only two devices and separate wires for sending and receiving, no identification or arbitration is needed. UART does not implement any error checking, so higher layers need to check for data integrity.
+
+### Bluetooth
+
+Bluetooth (BT) is a widely used wireless connection protocol developed for medium-bandwidth, low range applications. Different versions are available for low-energy, high-speed or basic needs. For many application layer protocols, the Serial Port Profile is mostly suited as it simulates a serial interface and is therfore interchangeable with UART or USB
```

Comment:
The serial port profile is only available in "old" Bluetooth, but not in BLE. For BLE (which is more common today, I guess) a custom profile can be created that provides similar behavior. Can you please add this information?